capa/features/freeze/features.py

```diff
@@ -278,6 +284,18 @@ class PropertyFeature(FeatureModel):
     description: Optional[str]
 
 
+class ReadPropertyFeature(PropertyFeature):
+    type: str = "property/read"
+    property: str = Field(alias="property/read")
+    description: Optional[str]
+
+
+class WritePropertyFeature(PropertyFeature):
+    type: str = "property/write"
+    property: str = Field(alias="property/write")
+    description: Optional[str]
```

Comment:
Is there something stopping us to use `PropertyFeature` throughout with a `r/w` attribute used for comparisons etc.?

Reply:
Taking this a step further - would it make sense to extend the [Feature](https://github.com/mandiant/capa/blob/81cb4b31e1025cdc14abdee13bdd92e4949c18dc/capa/features/common.py#L94-L146) class to support an "access" attribute similar to the `bitness` attribute that we previously supported (https://github.com/mandiant/capa/commit/55a5d10859e0a0e0d046c621bb4985faf9769e20).

where the "access" attribute is a simple enum e.g.:

```python
class Access(Enum):
    Read = 1
    Write = 2
    Unknown = 3
```

This would enable us to (optionally) extend the read/write specifier to all current features, including `Property`, and future features. This may even help reduce FPs e.g. the difference between `cmp` and `mov` using a `number` feature to detect PE parsing (we'd likely expect the number 0x5A4D to be read, not written, in this case) e.g.

```yaml
- number/read: 0x5A4D
```

We currently use `mnemonic` features to accomplish the above example but that doesn't easily allow for technology-agnostic rules e.g. a rule author wants to detect when the `number` 0x5A4D is read by both native PE and .NET executables.

Reply:
Hm, interesting idea. This would be a more generic alternative to the current `operand[n].number`/`operand[n].offset`. Let's open a new issue and discuss pros/cons, some examples, and details on the required changes and implications.

Reply:
I've removed the use of two different classes for capturing read/write property features. The `name` attribute of the `Property` class would be either `property/read` or `property/write`. This helps in emitting and displaying the read/write property without making any changes to the methods implemented in the `Feature` class. Although I did have to write an additional condition [here](https://github.com/mandiant/capa/blob/f8658b56c81dcda130288751a8b6dede880588ae/capa/rules.py#L577) since the `__init__` method for the `Property` class now takes an additional required argument. This may add some overhead. I would recommend that we implement the `access` attribute in the `Feature` class instead of the `Property` class. It would make the solution much cleaner.